lib/widgets/movie/movie_info.dart

```diff
@@ -25,48 +26,88 @@ class MovieInfo extends StatelessWidget {
+
+  @override
+  void initState() {
+    _movieGenres = provider.getGenresById(widget.movie.genres);
+    super.initState();
+  }
+
   @override
   Widget build(BuildContext context) {
     return Expanded(
       child: Padding(
-        padding: const EdgeInsets.all(padCard),
+        padding: const EdgeInsets.all(MovieInfo.padCard),
         child: Column(
           mainAxisSize: MainAxisSize.max,
           mainAxisAlignment: MainAxisAlignment.spaceEvenly,
           children: [
             CustomCard(
-              padding: padTitle,
+              padding: MovieInfo.padTitle,
               child: ListTile(
-                title: Text(movie.title),
-                subtitle: Text(movie.releaseDate),
+                title: Text(
+                  widget.movie.title,
+                ),
+                subtitle: Text(
+                  widget.movie.releaseDate,
+                ),
                 textColor: Theme.of(context).colorScheme.background,
               ),
             ),
             const SizedBox(
-              height: separation,
+              height: MovieInfo.separation,
             ),
             CustomCard(
-              padding: padGenres,
+              padding: MovieInfo.padGenres,
               child: Row(
                 children: [
                   Padding(
-                    padding: const EdgeInsets.all(padCard),
+                    padding: const EdgeInsets.all(MovieInfo.padCard),
                     child: Padding(
-                      padding: EdgeInsets.all(pad1),
+                      padding: EdgeInsets.all(widget.padGenresCard),
                       child: CircularPercentIndicator(
-                        percentIndicator: movie.score,
-                        radiusPercentIndicator: radiusPercentIndicator,
+                        percentIndicator: widget.movie.score,
+                        radiusPercentIndicator:
+                            MovieInfo.radiusPercentIndicator,
                       ),
                     ),
                   ),
                   Center(
                     child: SizedBox(
-                      width: textGenderWidth,
-                      child: Text(
-                        movie.genres,
-                        style: TextStyle(
-                            color: Theme.of(context).colorScheme.background),
-                        overflow: TextOverflow.visible,
+                      width: MovieInfo.textGenderWidth,
+                      child: FutureBuilder(
+                        future: _movieGenres,
+                        builder: (
+                          BuildContext context,
+                          AsyncSnapshot<dynamic> snapshot,
+                        ) {
+                          if (snapshot.connectionState ==
+                              ConnectionState.done) {
+                            if (snapshot.hasError) {
+                              throw Exception("Genres missing");
```

Comment:
I wouldn't throw an exception, as it closes the app! I'd show a `Text()` with a meaningful image, probably along with an alert dialog.

Reply:
Fixed, i create a CustomError class to allow me to show some images and for further needs